### DISCUSSION TITLE
Closing event extension for WinUIEx

### DISCUSSION THREAD

**Discussion by gmgallo** [dotMorten/WinUIEx#170] [Q&A]
The WinUI3 Window provides an after the fact Closed event, but is missing the Closing event like existed in the [Presentation Framework](https://learn.microsoft.com/en-us/dotnet/api/system.windows.window.closing?view=windowsdesktop-9.0). That gives you the hability to prevent a window to close at the wrong time, or asking the user for confirmation.

**Comment by dotMorten** [comment]
The AppWindow provides this: https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.windowing.appwindow.closing?view=windows-app-sdk-1.5

**Reply by gmgallo** [reply]
I missed that. Thanks
Actually, after some testing I found that Closed triggers before, not after the fact (inconsitent or counterintuitive name chosen?).
Adding a Closed event handler to the constructor or XAML you can do something like this:

```
    int count = 3;
    private void MyWindow_Closed(object sender, WindowEventArgs args)
    {
      if (--count != 0)
            args.Handled = true;
    }
```
This test makes you click three times the window close button before the window actually closes.
